src/com/opc_ua/opcua_objectstruct_helper.cpp

```diff
@@ -154,20 +135,15 @@ bool COPC_UA_ObjectStruct_Helper::addOPCUAStructTypeComponent(UA_Server *paServe
   UA_NodeId memberNodeId;
   if(paParentNodeId.identifierType == UA_NODEIDTYPE_STRING) {
     std::string memberBrowsePathStr = getStructMemberBrowsePath(paStructName, paStructMemberNameId);
-    char* memberBrowsePath = new char[memberBrowsePathStr.length() +1];
-    strncpy(memberBrowsePath, memberBrowsePathStr.c_str(), memberBrowsePathStr.length());
-    memberBrowsePath[memberBrowsePathStr.length()] = '\0';   
-    mStructTypeNames.push_back(memberBrowsePath);
-    memberNodeId = UA_NODEID_STRING(mOpcuaTypeNamespaceIndex, memberBrowsePath);
+    memberNodeId = UA_NODEID_STRING(mOpcuaTypeNamespaceIndex, memberBrowsePathStr.data());
   } else {
     memberNodeId = UA_NODEID_NUMERIC(mOpcuaTypeNamespaceIndex, 0);
   }
   UA_StatusCode status = UA_Server_addVariableNode(paServer, memberNodeId, paParentNodeId,
     UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
-    UA_QUALIFIEDNAME(mOpcuaTypeNamespaceIndex, memberName),
+    UA_QUALIFIEDNAME(mOpcuaTypeNamespaceIndex, &structMemberName[0]),
```

Comment:
I don't know if I am missing something here. In line 128 you are using `structMemberName.data()` instead of the start address like here. Is there a specific reason for this?

Reply:
It seems I was distracted that day. I unified all cases to be the same